ca/certificate-authority.go

```diff
@@ -277,18 +280,20 @@ func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest
 	commonName := ""
 	hostNames := make([]string, len(csr.DNSNames))
 	copy(hostNames, csr.DNSNames)
-	if len(csr.Subject.CommonName) > 0 {
-		commonName = strings.ToLower(csr.Subject.CommonName)
-		hostNames = append(hostNames, commonName)
-	} else if len(hostNames) > 0 {
-		commonName = strings.ToLower(hostNames[0])
-	} else {
-		err = core.MalformedRequestError("Cannot issue a certificate without a hostname.")
+	if len(csr.Subject.CommonName) > maxCNLength {
+		msg := fmt.Sprintf("Common name was longer than 64 bytes, was %d",
+			len(csr.Subject.CommonName))
+		err := core.MalformedRequestError(msg)
 		// AUDIT[ Certificate Requests ] 11917fa4-10ef-4e0d-9105-bacbe7836a3c
 		ca.log.AuditErr(err)
 		return emptyCert, err
 	}
 
+	if len(csr.Subject.CommonName) > 0 {
```

Comment:
This removes the code that hoists a SAN if there was no CN present in the CSR. That's good, and that's the plan, but I'd like to wrap it in a config flag. A single flag that covers both SAN hoisting and serial number insertion should be fine. And we don't need to flag-gate the 64-byte-CN rejection, since we definitely want to do that right out the gate.